spyder/plugins/updatemanager/workers.py

```diff
@@ -445,7 +480,150 @@ def start(self):
+                logger.debug(err, stack_info=True)
+
+    def _download_asset(self):
+        """Download Updater lock file"""
+        self.installer_path = osp.join(
+            get_temp_dir(),
+            "updates",
+            "spyder-updater",
+            str(self.asset_info["version"]),
+            self.asset_info["filename"],
+        )
+
+        if (
+            osp.exists(self.installer_path)
+            and validate_download(
+                self.installer_path, self.asset_info["checksum"]
+            )
+        ):
+            logger.debug(f"{self.installer_path} already downloaded.")
+            return
+
+        self._clean_installer_dir()
+        dirname = osp.dirname(self.installer_path)
+        os.makedirs(dirname, exist_ok=True)
+
+        url = self.asset_info["url"]
+        logger.info(f"Downloading {url} to {self.installer_path}")
+
+        page = requests.get(url, headers=GH_HEADERS)
+        page.raise_for_status()
+        with open(self.installer_path, 'wb') as f:
+            f.write(page.content)
+
+        if validate_download(self.installer_path, self.asset_info["checksum"]):
+            logger.info('Download successfully completed.')
+            with ZipFile(self.installer_path, 'r') as f:
+                f.extractall(dirname)
+        else:
+            raise UpdateDownloadError("Download failed!")
+
+    def _install_update(self):
+        """Install or update Spyder-updater environment."""
+        dirname = osp.dirname(self.installer_path)
+
+        if os.name == "nt":
+            plat = "win-64"
+        if sys.platform == "darwin":
+            plat = "osx-arm64" if platform.machine() == "arm64" else "osx-64"
+        else:
+            plat = "linux-64"
+        spy_updater_lock = osp.join(dirname, f"conda-updater-{plat}.lock")
+        spy_updater_conda = osp.join(dirname, "spyder-updater*.conda")
+
+        conda_exe = find_conda()
+        conda_cmd = "update" if UPDATER_VERSION > parse("0.0.0") else "create"
+        env_path = osp.join(osp.dirname(sys.prefix), "spyder-updater")
+        cmd = [
+            # Update spyder-updater environment
+            conda_exe,
+            conda_cmd, "--yes",
+            "--prefix", env_path,
+            "--file", spy_updater_lock,
+            "&&",
+            # Update spyder-updater
+            conda_exe,
+            "install", "--yes",
+            "--prefix", env_path,
+            "--no-deps",
+            "--force-reinstall",
+            spy_updater_conda
+        ]
+
+        logger.debug(f"""Conda command for the updater: '{" ".join(cmd)}'""")
+        proc = subprocess.run(
+            " ".join(cmd), shell=True, capture_output=True, text=True
+        )
+        proc.check_returncode()
+
+    def start(self):
+        """Main method of the worker."""
+        try:
+            self._check_asset_available()
+            if self.asset_info is None and UPDATER_VERSION == parse("0.0.0"):
+                proc.check_returncode()
```

Comment:
Seems like `proc` is not defined here?

Reply:
you are correct...I'll fix this. Thanks!